sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -213,6 +215,26 @@ case class DataSourceAnalysis(conf: SQLConf) extends Rule[LogicalPlan] with Cast
   }
 }
 
+/**
+ * Replace the V2 data source of table in [[InsertIntoTable]] to V1 [[FileFormat]].
+ * E.g, with temporary view `t` using [[FileDataSourceV2]], inserting into  view `t` fails
+ * since there is no correspoding physical plan.
+ * This is a temporary hack for making current data source V2 work.
```

Comment:
I don't like the idea of "temporary hack" rules. What is the long-term plan for getting rid of this? Or is this something we will end up supporting forever?

Reply:
This is definitely temporary. We have to add this, otherwise we can't migrate file data sources to V2.
I believe @cloud-fan have a plan for getting rid of this.